resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -79,23 +100,53 @@ public void onSuccess() {
 
     @Override
     public void onError(Throwable throwable) {
+        if (throwable instanceof TimeoutException) {
+            onTimeout();
+        } else {
+            onFailure(throwable);
+        }
+    }
+
+    private void onTimeout() {
+        timeouts.increment();
         if (!eventProcessor.hasConsumers()) {
             return;
         }
-        if (throwable instanceof TimeoutException) {
-            publishEvent(new TimeLimiterOnTimeoutEvent(name));
-        } else {
-            publishEvent(new TimeLimiterOnErrorEvent(name, throwable));
+        publishEvent(new TimeLimiterOnTimeoutEvent(name));
+    }
+
+    private void onFailure(Throwable throwable) {
+        errors.increment();
+        if (!eventProcessor.hasConsumers()) {
+            return;
         }
+        publishEvent(new TimeLimiterOnErrorEvent(name, throwable));
     }
 
     private void publishEvent(TimeLimiterEvent event) {
         try{
             eventProcessor.consumeEvent(event);
             LOG.debug("Event {} published: {}", event.getEventType(), event);
-        }catch (Throwable t){
+        } catch (Throwable t) {
             LOG.warn("Failed to handle event {}", event.getEventType(), t);
         }
     }
 
-}
+    public class TimeLimiterMetrics implements Metrics {
+
+        @Override
+        public long getNumberOfSuccessfulCalls() {
+            return successes.longValue();
+        }
+
+        @Override
+        public long getNumberOfErrorCalls() {
```

Comment:
Please rename to  `getNumberOfFailedCalls`, but only if we really measure metrics in the TimeLimiter itself.
I know it's confusing, because the public method is called `onError` and the event `TimeLimiterOnErrorEvent`, but I would like to use the same names like the CircuitBreaker uses.